src/compiler/utils/builder.js

```diff
@@ -59,5 +59,32 @@ export default {
 
   elTypes: {
     escapableRaw: ['textarea', 'title']
+  },
+
+  /**
+   * whether we are in an html_attribute given the current state
+   * NOTE: when we see a tornado body previous states are reset.
```

Comment:
IMO, inHTMLAttribute should be true no matter how deeply nested we are but the current `generateJS` step assumes we will create a new fragment for each tornado body. so that
```<div id="state1  {#section}state2<span id="state3 {#section2}state4{/section2}"></span>{/section"></div>```

we have 
state1 = HTML_ATTRIBUTE
state2 = TORNADO_BODY
state3=HTML_ATTRIBUTE
state4 = TORNADO_BODY
but IMO, in reality it should be

state1 = ['HTML_ATTRIBUTE']
state2 = ['HTML_ATTRIBUTE', 'TORNADO_BODY'] 
state3 = ['HTML_ATTRIBUTE', 'TORNADO_BODY', 'HTML_ATTRIBUTE'] 
state4 = ['HTML_ATTRIBUTE', 'TORNADO_BODY', 'HTML_ATTRIBUTE', 'TORNADO_BODY'] 

the calculation with lastIndex is to make this compatible with the existing generateJS but we could potentially do something smarter (throw an error that this is malformed) or optimize these to not create new fragments etc.